spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -40,6 +55,43 @@ <h1>[[DefineOwnProperty]] ( _P_, _Desc_ )</h1>
           1. Return *false*.
         </emu-alg>
       </emu-clause>
+      
+      <emu-clause id="sec-record-exotic-objects-hasproperty-p">
+        <h1>[[HasProperty]] ( _P_ )</h1>
+        <p>When the [[HasProperty]] internal method of a Record exotic object _R_ is called with property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is a Symbol, return *false*.
+          1. For each pair _p_ of _rec_.[[Fields]], do
+            1. Let _key_ be _p_.[[Key]]
+            1. If SameValue(_key_, _P_), return *true*.
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+            
+      <emu-clause id="sec-record-exotic-objects-get-p-receiver">
+        <h1>[[Get]] ( _P_, _Receiver_ )</h1>
+        <p>When the [[Get]] internal method of a Record exotic object _R_ is called with property key _P_ and ECMAScript language value _Receiver_, the following steps are taken:</p>
+        <emu-alg>
+          1. TODO
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-record-exotic-objects-set-p-receiver">
+        <h1>[[Set]] ( _P_, _Receiver_ )</h1>
+        <p>When the [[Set]] internal method of a Record exotic object _R_ is called with property key _P_ and ECMAScript language value _Receiver_, the following steps are taken:</p>
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-record-exotic-objects-delete-p">
+        <h1>[[Delete]] ( _P_ )</h1>
```

Comment:
Check the Delete method of module namespace exotic objects again

Reply:
yes, it should return true for non-existing keys. I just did a first pass for now but I'm going to do a second one

Reply:
Note, this is still a TODO